internal/resolve/catalog.go

```diff
@@ -101,6 +102,7 @@ func (r *CatalogResolver) Resolve(ctx context.Context, ext *ocv1alpha1.ClusterEx
 				return nil
 			}
 			if compare.ByVersion(*resolvedBundle, thisBundle) < 0 {
+				oldBundle = &thisBundle
```

Comment:
I'm not sure this approach covers all of the possibilities. I know the scenario talks about trying to "go backwards", but I think there's a similar scenario where the `spec.version` is ahead of the installed bundle, but contains no successors.

In general, I think we need to find a way to distinguish between the following:
- No bundles in this range
- The bundles in this range are not successors of the currently installed bundle

Distinguishing these may require more refactoring. Perhaps applying the various predicates separately would help us tell which predicates cause the resulting set of bundles to reduce to an empty list.